File: src/main/java/uk/laxd/dndSimulator/feature/generic/ExtraAttack.java
package uk.laxd.dndSimulator.feature.generic;

import uk.laxd.dndSimulator.character.Character;
import uk.laxd.dndSimulator.feature.Feature;
import uk.laxd.dndSimulator.feature.StaticFeature;

public class ExtraAttack extends StaticFeature {

    public ExtraAttack() {
        super("Extra Attack");
    }

    @Override
    public void onCreate(Character character) {
        character.setAttackCount(2);
    }
}

File: app/src/main/kotlin/uk/laxd/dndSimulator/feature/generic/ExtraAttack.kt
package uk.laxd.dndSimulator.feature.generic

import uk.laxd.dndSimulator.character.Character
import uk.laxd.dndSimulator.feature.StaticFeature
import uk.laxd.dndSimulator.character.CharacterClass

class ExtraAttack : StaticFeature("Extra Attack") {
    override fun onCreate(character: Character) {
        character.attackCount = 2
    }

    // TODO: Ranger/Fighter too
    override val characterClassRequired: CharacterClass = CharacterClass.BARBARIAN
    override val levelRequirement: Int = 5
}